### ISSUE TITLE
Password may not be sanitised as a string correctly - quotes are causing java errors and AUTH_FAILED

### ISSUE THREAD

**Issue by ghost** [schwabe/ics-openvpn#638]
# General information
1. Android 7.1.1 (ARM)
2. LineageOS 14.1 (Nightly)
3. Moto G (2013)
4. 0.6.64 (from F-Droid)

I am using a .ovpn config file with inline certificates.
# Description of the issue
When I had my password set to the following line, I had a massive output hundreds and hundreds of lines long with java errors, many lines saying a ".pn" file could not be found:

'BB'j.qEeE0-

Edit: I will also note that my user name had underscores in it. The user name was the default for my VPN service, so I do not feel comfortable sharing it, in case it can be traced back to me. I can share it via PM if needed.

**Comment by schwabe** [comment]
where did you set that password exactly?

**Comment by ghost** [comment]
@schwabe I am using a hosted VPN service (Protonmail's closed beta, ProtonVPN). I don't know if their server configuration files are available.

The same user/password combo works fine on OpenVPN via command line on two Debian desktop machines I use.

**Comment by schwabe** [comment]
Let me rephrase that: How do you input the password into the app. As part of the config file, when editing a profile or when prompted after selecting a VPN to connect.

**Comment by ghost** [comment]
I tried both, I had a file referred to in the configuration (auth-user-pass file ^1) and I tried manually entering the password into the configuration via your client.

I did not try using no saved password and entering it when prompted.

^1 file with the structure:
username
password

**Comment by schwabe** [comment]
hm there might something else wrong here, when I try your password I simply get a AUTH_FAILED from my server.

**Comment by ghost** [comment]
@schwabe Hmm. I tried with a bunch of different servers, thought there was some bug with the server or the android client. I only thought to change my user name and password to alphanumeric just before giving up.

If needed I can get the client config files and can try get the full log by changing my password back temporarily. I am heading to bed now and cannot do this until tomorrow.

**Comment by schwabe** [comment]
There might another bug, i.e. something in the config file that upsets the importer.

**Comment by ghost** [comment]
@schwabe Sorry for the delay.

.OVPN file:

http://paste.debian.net/hidden/f288c6c1/

filepass (user and pass have since been changed):

http://paste.debian.net/?hidden=761c9e33;lines=0

I could not find their server configuration, but if needed I can ask via the support team and I assume they would comply given that they bothered to license their configuration files.

**Comment by schwabe** [comment]
I just looked at this after years and the files are no longer available. So I cannot reproduce this anymore and closing it unresolved for now.
